pytest_gee/image_regression.py

```diff
@@ -40,9 +41,15 @@ def check(
             fullpath: complete path to use as a reference file. This option will ignore ``datadir`` fixture when reading *expected* files but will still use it to write *obtained* files. Useful if a reference file is located in the session data dir for example.
             scale: The scale to use for the thumbnail.
             viz_params: The visualization parameters to use for the thumbnail. If not given, the min and max values of the image will be used.
+            region:The region to use for clipping the image. If not given, the image's region will be used.
         """
         # rescale the original image
-        geometry = data_image.geometry()
+        if isinstance(region, (ee.FeatureCollection, ee.Feature)):
+            geometry = region.geometry()
+        elif isinstance(region, ee.Geometry):
+            geometry = region
+        else:
+            geometry = data_image.geometry()
```

Comment:
I would in this case let GEE deals with error handlingand be more naive on the type of inputs: 
```suggestion
        region = data_image if region is None else region 
        geometry = region if isinstance(region, ee.Geometry) else region.geometry()
```